src/coreclr/debug/createdump/crashinfo.cpp

```diff
@@ -328,24 +328,34 @@ CrashInfo::EnumerateMemoryRegionsWithDAC(MINIDUMP_TYPE minidumpType)
     {
         TRACE("EnumerateMemoryRegionsWithDAC: Memory enumeration STARTED (%d %d)\n", m_enumMemoryPagesAdded, m_dataTargetPagesAdded);
 
-        // Since on both Linux and MacOS all the RW regions will be added for heap
-        // dumps by createdump, the only thing differentiating a MiniDumpNormal and
-        // a MiniDumpWithPrivateReadWriteMemory is that the later uses the EnumMemory
-        // APIs. This is kind of expensive on larger applications (4 minutes, or even
-        // more), and this should already be in RW pages. Change the dump type to the
-        // faster normal one. This one already ensures necessary DAC globals, etc.
-        // without the costly assembly, module, class, type runtime data structures
-        // enumeration.
+        // Since on MacOS all the RW regions will be added for heap dumps by createdump, the
+        // only thing differentiating a MiniDumpNormal and a MiniDumpWithPrivateReadWriteMemory
+        // is that the later uses the EnumMemoryRegions APIs. This is kind of expensive on larger
+        // applications (4 minutes, or even more), and this should already be in RW pages. Change
+        // the dump type to the faster normal one. This one already ensures necessary DAC globals,
+        // etc. without the costly assembly, module, class, type runtime data structures enumeration.
+        CLRDataEnumMemoryFlags flags = CLRDATA_ENUM_MEM_DEFAULT;
         if (minidumpType & MiniDumpWithPrivateReadWriteMemory)
         {
+            // This is the old fast heap env var for backwards compatibility for VS4Mac.
             char* fastHeapDumps = getenv("COMPlus_DbgEnableFastHeapDumps");
             if (fastHeapDumps != nullptr && strcmp(fastHeapDumps, "1") == 0)
             {
                 minidumpType = MiniDumpNormal;
             }
+            // This the new variable that also skips the expensive (in both time and memory usage)
+            // enumeration of the low level data structures and adds all the loader allocator heaps
+            // instead. The above original env var didn't generate a complete enough heap dump on
+            // Linux and this new one does.
+            fastHeapDumps = getenv("COMPlus_EnableFastHeapDumps");
```

Comment:
Yes, I'll use clrconfignocache.h. I thought it would pull in utilcode or something messy, but it is self-contained.

I look at the EnableFastHeapDumps name as the new official (that should be documented) and *Dbg*EnableFastHeapDumps as just there for backwards compatibility for VS4Mac (and not be officially doc'ed). They can't be combined because the new one has some risk of incomplete dumps.